core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -437,27 +490,53 @@ private[spark] object SparkHadoopUtil extends Logging {
     // Note: this null check is around more than just access to the "conf" object to maintain
     // the behavior of the old implementation of this code, for backwards compatibility.
     if (conf != null) {
-      // Explicitly check for S3 environment variables
-      val keyId = System.getenv("AWS_ACCESS_KEY_ID")
-      val accessKey = System.getenv("AWS_SECRET_ACCESS_KEY")
-      if (keyId != null && accessKey != null) {
-        hadoopConf.set("fs.s3.awsAccessKeyId", keyId)
-        hadoopConf.set("fs.s3n.awsAccessKeyId", keyId)
-        hadoopConf.set("fs.s3a.access.key", keyId)
-        hadoopConf.set("fs.s3.awsSecretAccessKey", accessKey)
-        hadoopConf.set("fs.s3n.awsSecretAccessKey", accessKey)
-        hadoopConf.set("fs.s3a.secret.key", accessKey)
-
-        val sessionToken = System.getenv("AWS_SESSION_TOKEN")
-        if (sessionToken != null) {
-          hadoopConf.set("fs.s3a.session.token", sessionToken)
-        }
-      }
+      appendS3CredentialsFromEnvironment(hadoopConf, System.getenv)
```

Comment:
I'm still worrying about the chance of the regression of getting and passing a large map of strings here. There were some reports like this.
- https://github.com/apache/spark/pull/31244/files

Reply:
If you really want to hand over a map as a dependency injection, can you build a small one by reusing the existing logic only like `System.getenv("AWS_ACCESS_KEY_ID")`?

Reply:
ok

Reply:
having it pass down separate arguments; no map at all